python/tvm/relay/frontend/onnx.py

```diff
@@ -0,0 +1,1078 @@
+            'opset version {} of {} not implemented'.format(
+                version, cls.__name__))
+
+
+class Elemwise(OnnxOpConverter):
+    """ A helper class for elemwise op converters.
+    """
+    name = ''
+
+    @classmethod
+    def _impl_v1(cls, inputs, attr, params):
+        assert len(inputs) == 2, "Math op take 2 inputs, {} given".format(
+            len(inputs))
+        op_name = cls.name
+        conv_ops = ["conv2d", "conv2d_transpose"]
+        if attr.get('broadcast', 0) and any(x in str(inputs[0]) for x in conv_ops):
+            # TODO(zhreshold): remove hard coded infershape
+            axis = int(attr.get('axis', 0))
+            inputs[1] = _op.expand_dims(inputs[1], axis=axis, num_newaxis=2)
+        return get_relay_op(op_name)(*inputs)
+
+class Pool(OnnxOpConverter):
+    """ A helper class for pool op converters.
+    """
+    name = ''
+
+    @classmethod
+    def _impl_v1(cls, inputs, attr, params):
+        return AttrCvt(
+            op_name=dimension_picker(cls.name),
+            transforms={
+                'kernel_shape': 'pool_size',
+                'pads': ('padding', (0, 0), revert_caffe2_pad)
+            },
+            # very weird attributes here in onnx, force check
+            ignores=['dilations'],
+            # TODO(zhreshold): make sure ceil_mode in onnx, and layout?
+            extras={'ceil_mode': False},
+            custom_check=dimension_constraint())(inputs, attr, params)
+
+
+class Absolute(OnnxOpConverter):
+    """ Operator converter for Absolute.
+    """
+
+    @classmethod
+    def _impl_v1(cls, inputs, attr, params):
+        return _op.nn.relu(inputs[0]) + _op.nn.relu(_op.negative(inputs[0]))
+
+
+class Add(Elemwise):
+    """ Operator converter for Add.
+    """
+    name = 'add'
+
+
+class AveragePool(Pool):
+    """ Operator converter for AveragePool.
+    """
+    name = 'avg_pool'
+
+
+class BatchNorm(OnnxOpConverter):
+    """ Operator converter for BatchNorm.
+    """
+
+    @classmethod
+    def _impl_v1(cls, inputs, attr, params):
+        # TODO(zhreshold): 'spatial' is not properly handled here.
+        out = AttrCvt(
+            op_name='batch_norm',
+            ignores=['spatial', 'is_test', 'consumed_inputs', 'momentum'])(inputs, attr,
+                                                                           params)
+        return out[0]
+
+
+class Conv(OnnxOpConverter):
+    """ Operator converter for Conv.
+    """
+
+    @classmethod
+    def _impl_v1(cls, inputs, attr, params):
+        # get number of channels
+        out = AttrCvt(op_name=dimension_picker('conv'),
+                      transforms={
+                          'kernel_shape': 'kernel_size',
+                          'dilations': ('dilation', (0, 0)),
+                          'pads': ('padding', (0, 0), revert_caffe2_pad),
+                          'group': ('groups', 1)},
+                      custom_check=dimension_constraint())(inputs, attr, params)
```

Comment:
I think we shouldn't pass 3 inputs (if bias available).

Reply:
will strip to first 2 inputs